### PR TITLE
Simplify mnnist example using Gluon API and update README

### PR DESCRIPTION
**Pull request by apeforest**
Current mxnet examples uses an old module API which has been gradually replaced by a more user-friendly Gluon API. Replacing the example and README using Gluon API for mxnet.

### CODE REVIEW

**Comment by apeforest** [comment]
@yuxihu @eric-haibin-lin Please help to review. Thanks!

File: examples/mxnet_mnist.py [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
                     help='number of training epochs (default: 5)')
-parser.add_argument('--lr', type=float, default=0.05,
-                    help='learning rate (default: 0.05)')
-parser.add_argument('--momentum', type=float, default=0.5,
-                    help='SGD momentum (default: 0.5)')
+parser.add_argument('--lr', type=float, default=0.01,
+                    help='learning rate (default: 0.01)')
+parser.add_argument('--momentum', type=float, default=0.9,
+                    help='SGD momentum (default: 0.9)')
+parser.add_argument('--no-cuda', action='store_true', default=True,
```
Comment (yuxihu) [inline examples/mxnet_mnist.py]: how to set no_cuda to False?

Comment (apeforest) [reply examples/mxnet_mnist.py]: Good catch. Changed argument to -use-gpu

**Comment by alsrgv** [comment]
How about using something like this: https://github.com/horovod/horovod/commit/a5fb7a215a8a62cd5b8862fa819f6371050321d9#diff-10729f4cf94906d5762883f7163f91dbR17 & https://github.com/horovod/horovod/commit/a5fb7a215a8a62cd5b8862fa819f6371050321d9#diff-10729f4cf94906d5762883f7163f91dbR27 ?

File: examples/mxnet_mnist.py [hunk 2]
```diff
@@ -1,29 +1,36 @@
...
+import time
 
-import horovod.mxnet as hvd
 import mxnet as mx
+import horovod.mxnet as hvd
+from mxnet import autograd, gluon, nd
 from mxnet.test_utils import download
 
 # Training settings
 parser = argparse.ArgumentParser(description='MXNet MNIST Example')
```
Comment (eric-haibin-lin) [inline examples/mxnet_mnist.py]: This may be a general question for all horovod examples: 
There are a lot of examples for ResNet on ImageNet. This is great. But I do not find instructions on how to run those example (e.g. with x learning rate and y batch size on z GPUs this example converges to 75.5% accuracy). This will help people validate that everything is working when they try out the example.

File: README.md [hunk 3]
```diff
@@ -214,15 +214,10 @@ Horovod supports MXNet and regular TensorFlow in similar ways.
...
-
-###### Gluon API
 ```python
-from mxnet import autograd, gluon
 import mxnet as mx
 import horovod.mxnet as hvd
+from mxnet import autograd, gluon
 
 # Initialize Horovod
 hvd.init()
```
Comment (eric-haibin-lin) [inline README.md]: Is the parameter hook for gluon already done? As we discussed last time, if the shape of a parameter has to be inferred after a batch of data is seen, the parameter will not be broadcast. Therefore, the correctness of the program depends on whether all workers are initialized with the same random seed. Maybe we should call it out as a known issue if it is not yet fixed.

Comment (yuxihu) [reply README.md]: Not yet. I will look into it soon.

File: examples/mxnet_mnist.py (hunk 2 above)
Comment (apeforest) [reply examples/mxnet_mnist.py]: @eric-haibin-lin The default values are the ones we used to achieve 75.5% reported. Users can reproduce this using the defalt argument values.

File: README.md (hunk 3 above)
Comment (eric-haibin-lin) [reply README.md]: I created an issue in case others it affects others: https://github.com/horovod/horovod/issues/895

Comment (yuxihu) [reply README.md]: @eric-haibin-lin I think you should mention Gluon in the title of the issue. It does not affect Module API.
